### ISSUE TITLE
git: parse_git_dirty is too slow

### ISSUE THREAD

**Issue by robertfoss** [robbyrussell/oh-my-zsh#5066]
When opening folders that contain large git repositories, like the linux-kernel, git_prompt_info and parse_git_dirty take >3 seconds to run.

Waiting >3  is pretty much unacceptable.
Either adding a flag for disabling slow git functionality or removing slow functionality altogether are acceptable alternatives to me.

Currently I'm using a work-around: https://gist.github.com/msabramo/2355834
But since oh-my-zsh auto-updates over git I'm going to have merge conflicts/my changes disregarded.

**Comment by ccharlton** [comment]
Ditto.

**Comment by ccharlton** [comment]
I'd be fine with even having it's update frequency not be every second, but every XX seconds.

**Comment by Braunson** [comment]
Having the same issue, there's < 30 changes in my Git Repo :(

**Comment by ccharlton** [comment]
Yep, my change list is usually fairly small too so this would be nice to get boosted.

**Comment by iainp999** [comment]
I'm having the same issue, again with a small number of local changes in the git repo.

Made local modifications to workaround for now but using a plugin manager, so changes would be wiped out on next update :(

**Comment by davidhcefx** [comment]
You can limit the git command via a timeout:
https://github.com/ohmyzsh/ohmyzsh/blob/f6bc949c05fd18ee557a89c92f82ff556200707d/lib/git.zsh#L6-L11

```diff
- GIT_OPTIONAL_LOCKS=0 command git "$@"
+ GIT_OPTIONAL_LOCKS=0 command timeout 1s git "$@"
```

**Comment by mcornella** [comment]
Note that you can disable `parse_git_dirty` per-repository with `git config oh-my-zsh.hide-dirty 1`, or making that global to disable it for all repositories.

But yeah, it should be made faster.

**Comment by basicdays** [comment]
Having this issue as well. @mcornella idea of the git config option to disable this feature works as a workaround on my end though thankfully.

**Comment by jingxixu** [comment]
I am having the same issue and @mcornella 's solution of git config works. But I would like to understand a bit more about why this works. How is the option `oh-my-zsh.hide-dirty` used? Does this mean nothing related to git is run on this repository? If yes, why is my terminal still showing the branch `[master]`?

**Comment by shmish111** [comment]
Thanks @mcornella I followed the tip of disabling checking for dirty on a big repo, I'm not really bothered as I usually check the status manually anyway, I had actually forgotten the little `#` symbol at the end of the line anyway so clearly I don't need it!

**Comment by ejoubaud** [comment]
People coming across this are also likely to want to add:

```
git config --add oh-my-zsh.hide-status 1
```

See https://unix.stackexchange.com/questions/565905/oh-my-zshs-prompt-is-slow-how-to-fix-this
